downloads/swan-lake-release-notes/swan-lake-2201.2.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,759 @@
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    io:println(function:call(getSum)); // 30
+    io:println(function:call(getSum, 10)); // 40
+    io:println(function:call(getSum, 10, 15)); // 45
+    io:println(function:call(getSum, 10, 15, 25)); // 50
+}
+
+function getSum(int a = 0, int b = 10, int c = 20) returns int {
+    return a + b + c;
+}
+```
+
+If the arguments do not match the parameters expected by the function, the function call results in a panic.
+
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    callFn(getSum); // The `function:call` call results in a panic.
+}
+
+function callFn(function fn) {
+    any|error value = function:call(fn, 10);
+    io:println(value);
+}
+
+function getSum(int a, int b, int c) returns int {
+    return a + b + c;
+}
+```
+
+#### Bug fixes
+
+- A bug that resulted in allowing access to non-public error intersection types outside the module has been fixed.
+
+```ballerina
+public type E1 distinct error;
+ 
+public type E2 distinct error;
+ 
+type E3 distinct error;
+ 
+type E4 distinct error;
+ 
+type E5 E1 & E2; // not allowed to access outside the module
+ 
+type E6 E3 & E4; // not allowed to access outside the module
+```
+
+- Fixed a bug that resulted in a compilation error not being logged for invalid usage of an on-conflict clause in a query with inner queries.
+
+```ballerina
+type Token record {|
+    readonly int idx;
+    string value;
+|};
+
+type TokenTable table<Token> key(idx);
+
+public function main() {
+    Token[] tbl = from Token i in (table key(idx) from var j in [1, 2]
+                        select {
+                            idx: j,
+                            value: "A"
+                        })
+                    select {
+                        idx: i.idx,
+                        value: "A"
+                    }
+                    on conflict error("Duplicate Key"); // Now an error
+}
+```
+
+- A bug that resulted in a compilation error not being logged when a spread field is used in a table member to specify a field that is part of the key sequence has been fixed. For every field name in the key sequence of a table, a mapping constructor member must have a specific field corresponding to the field.
+
+```ballerina
+type Employee record {
+    readonly int id;
+    string name;
+    string dept;
+};
+
+public function main() {
+    Employee employee = {
+        id: 1234,
+        name: "Amy Wilson",
+        dept: "legal"
+    };
+
+    table<Employee> key(id) employees = table [
+        // Results in an error now.
+        {...employee}
+    ];
+}
+```
+
+To view bug fixes, see the [GitHub milestone for 2201.2.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%22Ballerina+2201.2.0%22).
```

Comment:
Is this link correct? @gimantha / @praneesha ?

Reply:
It should be https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%222201.2.0%22

Reply:
@anupama-pathirage Yes.. at this moment, the commented url is the correct one. But aren't we changing the milestones once released? All other Previous milestones are prefixed with "Ballerina"

Reply:
@NipunaMadhushan Any thoughts?

Reply:
@dulajdilshan @keizer619 - Please confirm if the link is correct.

Reply:
The commented link shows the correct issues tagged for the milestone. Are we changing the milestone after the release to Ballerina 2201.2.0?

Reply:
We are not changing milestones as we release. I think we can keep just the without "Ballerina" part

Reply:
Updated to https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%222201.2.0%22.